src/SurfaceSchemes/surface_schemes.jl

```diff
@@ -0,0 +1,233 @@
+    ustar = FT(0.28) # 0.28 is taken from Bomex. TODO: Approximate from LES TKE.
+    thermo_params = CAP.thermodynamics_params(params)
+    ts = TD.PhaseNonEquil_pθq(thermo_params, p, θ, TD.PhasePartition(q_vap))
+    T = TD.air_temperature(thermo_params, ts)
+    shf = FT.(Dierckx.Spline1D(t_data, shf_data; k = 1))
+    lhf = FT.(Dierckx.Spline1D(t_data, lhf_data; k = 1))
+    # TODO: Replace Dierckx with a type-stable interpolation and remove the FT.
+    function surface_state(surface_coordinates, interior_z, t)
+        fluxes = HeatFluxes(; shf = shf(t), lhf = lhf(t))
+        parameterization = Roughness(z0; fluxes, ustar)
+        return SurfaceState(; parameterization, T, p, q_vap)
+    end
+    return surface_state
+end
+
+struct DYCOMS_RF01 end
+function (::DYCOMS_RF01)(params)
+    FT = eltype(params)
+    T = FT(292.5)
+    p = FT(101780)
+    q_vap = FT(0.01384) # 0.01384 is taken from Pycles. TODO: Compute qstar(T).
+    z0 = FT(1e-4)
+    shf = FT(15)
+    lhf = FT(115)
+    parameterization = Roughness(z0; fluxes = HeatFluxes(; shf, lhf))
+    return SurfaceState(; parameterization, T, p, q_vap)
+end
+
+struct DYCOMS_RF02 end
+function (::DYCOMS_RF02)(params)
+    FT = eltype(params)
+    T = FT(292.5)
+    p = FT(101780)
+    q_vap = FT(0.01384) # 0.01384 is taken from Pycles. TODO: Compute qstar(T).
+    z0 = FT(1e-4)
+    shf = FT(16)
+    lhf = FT(93)
+    ustar = FT(0.25)
+    parameterization = Roughness(z0; fluxes = HeatFluxes(; shf, lhf), ustar)
+    return SurfaceState(; parameterization, T, p, q_vap)
+end
+
+struct Rico end
+function (::Rico)(params)
+    FT = eltype(params)
+    thermo_params = CAP.thermodynamics_params(params)
+    T_surface = FT(299.8)
+    p_surface = FT(101540)
+    z0 = FT(1.5e-4)
+    Cd = FT(0.001229)
+    Ch = FT(0.001094)
+    Cq = FT(0.001133) # TODO: Add support for Cq to SF.Coefficients.
+    # Saturated surface conditions for a given surface temperature and pressure
+    p_sat_surface =
+        TD.saturation_vapor_pressure(thermo_params, T_surface, TD.Liquid())
+    ϵ_v = TD.Parameters.R_d(thermo_params) / TD.Parameters.R_v(thermo_params)
+    q_surface = ϵ_v * p_sat_surface / (p_surface - p_sat_surface * (1 - ϵ_v))
+
+    function surface_state(surface_coordinates, interior_z, t)
+        # Adjust the coefficients from 20 m to the actual value of z.
+        adjustment = (log(20 / z0) / log(interior_z[1, 1, 1, 1, 1] / z0))^2
+        parameterization =
+            ExchangeCoefficients(Cd = Cd * adjustment, Ch = Ch * adjustment)
+        SurfaceState(;
+            parameterization,
+            T = T_surface,
+            p = p_surface,
+            q_vap = q_surface,
+        )
+    end
+    return surface_state
+end
+
+struct TRMM_LBA end
+function (::TRMM_LBA)(params)
+    FT = eltype(params)
+    T = FT(273.15) + FT(23.7)
+    p = FT(99130)
+    q_vap = FT(0.02245)
+    z0 = FT(0) # 0 seems to be here at random. TODO: Should this be 1e-4?
```

Comment:
I'm surprised that it works. Maybe z0 is never used in this case?

Reply:
It is used for SurfaceFluxes, the comment is just an old artifact

Reply:
hmm, I think you will have a divided-by-zero problem if this value is actually used in SurfaceFluxes. So maybe it is required but not used? Anyway, not a big issue.

Reply:
Should we change it to be a NaN and add  a comment explaining that it isn't used?

Reply:
I think that would be better.